core/integration-test/src/core.test.js

```diff
@@ -73,8 +72,8 @@ describe("Core", () => {
     const originalGetResponse = await request(URL)
       .get("/pipelines/" + pipelineId);
 
-    let updatedConfig = pipelineConfig;
-    updatedConfig.adapter.location = "http://www.disrespect.com";
+    let updatedConfig = Object.assign({}, pipelineConfig);
+    updatedConfig.datasourceId = 999;
```

Comment:
Pipeline references the underlying datasource just by id. No validation that this datasource exists yet.